### DISCUSSION TITLE
train_gpt_demo.py例子中的一些疑问

### DISCUSSION THREAD

**Discussion by yhcc** [hpcaitech/ColossalAI#2182] [Community | Q&A] [answered]
hello，能麻烦帮我解答下关于在ZERO中使用tensor_parallelize的疑问吗？
（1）在https://github.com/hpcaitech/ColossalAI/blob/65f56f49e8ff5db4fe9b1911f90f8d1950fe1c47/examples/language/gpt/train_gpt_demo.py#L169-L174
这个例子中，似乎只需要对column进行切分，但这样其实是会导致在每个卡上计算出来的tensor是在最后一个维度为1/n_proc原始大小，这里是CAI会做特定的处理让它参与运算后返回的activation是已经gather过，所以大小也是expected的大小了吗？
（2）对应的https://github.com/hpcaitech/ColossalAI/blob/65f56f49e8ff5db4fe9b1911f90f8d1950fe1c47/examples/language/gpt/train_gpt_demo.py#L157-L168
这个161行意思是就是指因为后续还有一个根据row切分的，所以暂时不需要gather回未切分前tensor的大小么？

**Comment by ver217** [comment, marked as answer]
目前为了兼容性是都会自动gather output的

**Reply by yhcc** [reply]
感谢回答。
